nmdc_schema/nmdc_data.py

```diff
@@ -59,56 +73,93 @@ def get_materialized_nmdc_yaml_string():
     return materialized_nmdc_yaml_string
 
 
-def get_nmdc_jsonschema_bytesIO() -> io.BytesIO:
+def get_nmdc_jsonschema_bytesIO(variant: Optional[SchemaVariantIdentifier] = None) -> io.BytesIO:
     """Returns the nmdc.schema.json file as bytes steam.
     This function is not intended to be used directly, but it used by other functions
 
     Returns
     -------
     BytesIO
         A bytes stream of nmdc.schema.json file.
+
+    >>> stream_a = get_nmdc_jsonschema_bytesIO()
+    >>> type(stream_a) is io.BytesIO
+    True
+    >>> stream_b = get_nmdc_jsonschema_bytesIO(variant=SchemaVariantIdentifier.nmdc_materialized_patterns)
+    >>> type(stream_b) is io.BytesIO
+    True
     """
-    # get nmdc.yaml file from the package data
-    return io.BytesIO(pkgutil.get_data(__name__, "nmdc.schema.json"))
+
+    # Determine which file we will use.
+    file_name = "nmdc.schema.json"
+    if variant == SchemaVariantIdentifier.nmdc_materialized_patterns.value:
```

Comment:
I'm surprised you need `.value` here. The appeal of subclassing `str` and `Enum` is that you should be able to directly compare string and `SchemaVariantIdentifier` values.

Reply:
Thanks for bringing this up. The `.value` part was optional. I opted to [include/add it](https://github.com/microbiomedata/berkeley-schema-fy24/pull/204/commits/36c24d20e7d9d0e7da6aedade39a961209fca3d6) last night after reading the code without it, because I was having trouble keeping Python's behavior with respect to `Enum`s straight and wanted to eliminate the question of whether the value on the right-hand side of the `==` was the value of the `Enum`'s attribute or some other thing (e.g. the _name_ of that attribute). I think I made it ambiguous by using identical strings in both places, in the `Enum` definition above.

Anyway, I go through this every time I use Python enums—I'll make a point to familiarize myself with them more.

For this PR, I have removed the `.value`.